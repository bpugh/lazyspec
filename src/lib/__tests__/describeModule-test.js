const fs = require('fs');

const describeModule = require('../describeModule');
const parseModule = require('../parseModule');

function expectDescription(moduleName) {
  const unitPath = `spec/fixtures/${moduleName}.js`;
  const src = fs.readFileSync(unitPath);
  const ast = parseModule(unitPath, src);
  const description = describeModule(moduleName, ast);

  expect(description).toMatchSnapshot();
}

describe('describeModule', () => {
  it('should describe simple exports', () => {
    expectDescription('esExports');
  });

  it('should describe an ES5 component', () => {
    expectDescription('CreateClass');
  });

  xit('should describe an ES6 component', () => {
    expectDescription('DefaultableComponent');
  });

  xit('should describe an ES6 with Flow component', () => {
    expectDescription('FlowComponent');
  });
});
