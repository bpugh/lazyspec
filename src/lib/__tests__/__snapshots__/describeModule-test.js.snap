exports[`describeModule should describe a createClass component 1`] = `
Object {
  "description": "",
  "exportsDefault": true,
  "hasExports": true,
  "methods": Array [
    Object {
      "docblock": null,
      "modifiers": Array [],
      "name": "_setup",
      "params": Array [],
      "returns": null,
    },
    Object {
      "docblock": null,
      "modifiers": Array [],
      "name": "show",
      "params": Array [],
      "returns": null,
    },
  ],
  "moduleName": "CreateClass",
  "namedExportNames": Array [],
  "props": Object {
    "disabled": Object {
      "description": "",
      "required": false,
      "type": Object {
        "name": "bool",
      },
    },
    "height": Object {
      "description": "",
      "required": false,
      "type": Object {
        "name": "number",
      },
    },
    "label": Object {
      "description": "",
      "required": false,
      "type": Object {
        "name": "string",
      },
    },
    "level": Object {
      "description": "",
      "required": false,
      "type": Object {
        "name": "enum",
        "value": Array [
          Object {
            "computed": false,
            "value": "\"primary\"",
          },
          Object {
            "computed": false,
            "value": "\"secondary\"",
          },
          Object {
            "computed": false,
            "value": "\"tertiary\"",
          },
          Object {
            "computed": false,
            "value": "\"quaternary\"",
          },
          Object {
            "computed": false,
            "value": "\"text\"",
          },
          Object {
            "computed": false,
            "value": "\"icon\"",
          },
        ],
      },
    },
  },
}
`;

exports[`describeModule should describe simple exports 1`] = `
Object {
  "exportsDefault": true,
  "hasExports": true,
  "moduleName": "esExports",
  "namedExportNames": Array [
    "refConst",
    "refFunc",
    "inlineConst",
    "inlineFunc",
  ],
}
`;
